nixos/modules/services/web-apps/funkwhale/funkwhale.nix

```diff
@@ -0,0 +1,329 @@
+          '';
+        };
+        funkwhale-init = {
+          description = "Funkwhale initialization";
+          after = [ "funkwhale-psql-init.service" ];
+          wantedBy = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          before   = [ "funkwhale-server.service" "funkwhale-worker.service" "funkwhale-beat.service" ];
+          environment = funkwhaleEnv;
+          script = ''
+            chmod a+rx ${funkwhaleHome}
+            if ! test -e ${cfg.api.media_root}; then
+            mkdir -p ${cfg.api.media_root}
+            mkdir -p ${cfg.api.static_root}
+            mkdir -p ${cfg.music_directory_path}
+            echo "#!/bin/sh
+            
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py createsuperuser" > ${funkwhaleHome}/createSuperUser.sh
+            chmod u+x ${funkwhaleHome}/createSuperUser.sh
+            chown -R funkwhale.users ${funkwhaleHome}
+
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py migrate
+            ${pythonEnv}/bin/python ${pkgs.funkwhale}/manage.py collectstatic
+            fi
+            if ! test -e ${funkwhaleHome}/config; then
+              mkdir -p ${funkwhaleHome}/config
+              ln -s ${funkwhaleEnvFile} ${funkwhaleHome}/config/.env
+              ln -s ${funkwhaleEnvFile} ${funkwhaleHome}/.env
+            fi
+          '';
+        };
+
+        funkwhale-server = {
+          description = "Funkwhale application server";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = "${pythonEnv}/bin/daphne -b ${cfg.api_ip} -p ${cfg.api_port} config.asgi:application --proxy-headers";
+
+          wantedBy = [ "multi-user.target" ];
+        };
+
+        funkwhale-worker = {
+          description = "Funkwhale celery worker";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = ''
+            if ! test -e ${funkwhaleHome}/run; then
+            mkdir -p ${funkwhaleHome}/run
+            fi
+            ${pythonEnv}/bin/celery -A funkwhale_api.taskapp worker -l INFO
+            '';
+
+          wantedBy = [ "multi-user.target" ];
+        };
+
+        funkwhale-beat = {
+          description = "Funkwhale celery beat process";
+          partOf = [ "funkwhale.target" ];
+
+          serviceConfig = serviceConfig;
+          environment = funkwhaleEnv;
+          script = ''
+            if ! test -e ${funkwhaleHome}/run; then
+            mkdir -p ${funkwhaleHome}/run
+            fi
+            ${pythonEnv}/bin/celery -A funkwhale_api.taskapp beat -l INFO --schedule="${funkwhaleHome}/run/celerybeat-schedule.db"  --pidfile="${funkwhaleHome}/run/celerybeat.pid"
+            '';
+
+          wantedBy = [ "multi-user.target" ];
+        };
+
+      };
+
+  };
+
+  meta = {
+    maintainers = with lib.maintainers; [ mmai ];
```

Comment:
Fill in the rest of the meta info, look at other packages.

Reply:
You mean the "doc" option ? There is nothing else I can put for a module according to https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/meta.nix. The meta info in pkgs/servers/web-apps/funkwhale/default.nix are correctly set I think.

Reply:
Oh sorry, I mixed it up with the package declaration, my fault.